Apps/Sandcastle/gallery/development/3D Models Instancing.html

```diff
@@ -0,0 +1,267 @@
+        }).otherwise(function(error){
+            window.alert(error);
+        });
+    }
+
+    model.readyPromise.then(function(model) {
+        var boundingSphere = new Cesium.BoundingSphere();
+        Cesium.BoundingSphere.fromPoints(points, boundingSphere);
+        orientCamera(boundingSphere.center, boundingSphere.radius + model.boundingSphere.radius);
+    });
+}
+
+function reset() {
+    scene.primitives.removeAll();
+    var instances = [];
+    var gridSize = Math.sqrt(count);
+
+    for (var y = 0; y < gridSize; ++y) {
+        for (var x = 0; x < gridSize; ++x) {
+            var longitude = centerLongitude + spacing * (x - gridSize/2);
+            var latitude = centerLatitude + spacing * (y - gridSize/2);
+            var position = Cesium.Cartesian3.fromDegrees(longitude, latitude, height);
+
+            var heading = Math.random();
+            var pitch = Math.random();
+            var roll = Math.random();
+            var scale = (Math.random() + 1.0)/2.0;
+            var color = Cesium.Color.fromRandom();
+            var show = (Math.random() > 0.5);
+
+            var modelMatrix = Cesium.Transforms.headingPitchRollToFixedFrame(position, heading, pitch, roll);
+            Cesium.Matrix4.multiplyByUniformScale(modelMatrix, scale, modelMatrix);
+
+            instances.push({
+                modelMatrix : modelMatrix,
+                color : color,
+                show : show
+            });
+        }
+    }
+
+    useCollection ? createCollection(instances) : createModels(instances);
+}
+
+Sandcastle.addToolbarMenu([ {
+    text : '1024 instances',
+    onselect : function() {
+        count = 1024;
+        reset();
+    }
+}, {
+    text : '100 instances',
+    onselect : function() {
+        count = 100;
+        reset();
+    }
+}, {
+    text : '25 instances',
+    onselect : function() {
+        count = 25;
+        reset();
+    }
+}, {
+    text : '4 instances',
+    onselect : function() {
+        count = 4;
+        reset();
+    }
+}, {
+    text : '10000 instances',
+    onselect : function() {
+        count = 10000;
+        reset();
+    }
+}]);
+
+Sandcastle.addToolbarMenu([ {
+    text : 'Aircraft',
+    onselect : function() {
+        url = '../../SampleData/models/CesiumAir/Cesium_Air.bgltf';
+        spacing = 0.0002;
+        reset();
+    }
+}, {
+    text : 'Ground vehicle',
+    onselect : function() {
+        url = '../../SampleData/models/CesiumGround/Cesium_Ground.bgltf';
+        spacing = 0.00008;
+        reset();
+    }
+}, {
+    text : 'Milk truck',
+    onselect : function() {
+        url = '../../SampleData/models/CesiumMilkTruck/CesiumMilkTruck.bgltf';
+        spacing = 0.00008;
+        reset();
+    }
+}, {
+    text : 'Skinned character',
+    onselect : function() {
+        url = '../../SampleData/models/CesiumMan/Cesium_Man.bgltf';
+        spacing = 0.00001;
+        reset();
+    }
+}]);
+
+
+Sandcastle.addToolbarMenu([ {
+    text : 'Instancing Enabled',
```

Comment:
@lilleyse did you look at this performance issue?  I do not want to merge this until we track it down.